DIPs/DIP1005.md

```diff
@@ -0,0 +1,257 @@
+
+The inline imports don't have priority over existing `import`s visible to the declaration. The lookup is equivalent to placing the inline `import`s in the enclosing scope of the declaration, where they participate in name resolution per the current language rules.
+
+### Examples
+
+Below are a few examples taken from the standard library:
+
+```d
+auto uninitializedArray(T, I...)(I sizes) nothrow @system
+if (isDynamicArray!T && allSatisfy!(isIntegral, I) &&
+    hasIndirections!(ElementEncodingType!T))
+import (std.meta, std.range, std.traits)
+{
+    ...
+}
+```
+
+Alternatively, the declaration may specify the exact symbols needed by using multiple imports:
+
+```d
+auto uninitializedArray(T, I...)(I sizes) nothrow @system
+if (isDynamicArray!T && allSatisfy!(isIntegral, I) &&
+    hasIndirections!(ElementEncodingType!T))
+import (std.meta : allSatisfy)
+import (std.range : ElementEncodingType)
+import (std.traits : hasIndirections, isDynamicArray, isIntegral)
+{
+    ...
+}
+```
+
+#### Workarounds and Alternatives
+
+The obvious workaround to the problem that dependencies must be module-level is to simply define many small modules---in the extreme, one per declaration. Each such small module would import the modules on which that declaration depends. For convenience, `package.d` modules may be provided to aggregate several modules. This approach has the following tradeoffs:
+
+* Library authors face a tension between organizing code in logical units pertaining to the problem domain, and organizing code according to low-level dependency details. They will also be forced to routinely navigate large file hierarchies with many files, which may not be the preferred project organization.
+* Client code must choose between using detailed import lists or convenient `package.d` imports.
+    * If convenient grouped imports are used, the advantage of fine-grained dependency control is lost.
+    * If detailed import lists are used, they are verbose and must be updated often. Because it is not an error to not use an imported symbol, over time the import lists will become a large set including the actually needed set, thus eroding the advantage of the approach in the first place. Special tooling and maintenance tasks would be needed to remove unneeded imports once in a while.
+
+We consider such a workaround nonscalable and undesirable for most projects. At the same time we note that this proposal does not preclude such a project organization.
+
+There are a number of alternative approaches that have been (and some still are) considered.
+
+* Full lazy `import`s. Assume all `import`s are lazy without any change in the language. That would allow an idiom by which libraries use fully qualified names everywhere, and the `import`s would be effected only when names are looked up. This would allow scalability but not the dependency-carrying aspect.
```

Comment:
Ok, I see your point and you are correct, but it still looks wrong. Maybe use "actualized" or "realized"?